clientapi/routing/sendevent.go

```diff
@@ -35,6 +36,10 @@ type sendEventResponse struct {
 	EventID string `json:"event_id"`
 }
 
+var (
+	mutexes sync.Map // (roomID+userID) -> mutex. mutexes to ensure correct ordering of sendEvents
```

Comment:
Since this will ultimately become a global within the package, perhaps we can name it more descriptively to reduce the chance of collision? Maybe `userRoomSendMutexes` or something.